### PR TITLE
working on filtering

### PR DESCRIPTION
**Pull request by kd2718**
this is my first MR writing rust, so there are a few things i am not sure about. I am not yet finished.

### CODE REVIEW

**Comment by kd2718** [comment]
image

image


Added filter buttons to clean up tournament view.

File: shared_types/src/lib.rs [hunk 1]
```diff
@@ -41,3 +42,4 @@ pub use tournament_details::TournamentDetails;
 pub use tournament_game_result::TournamentGameResult;
 pub use tournament_mode::TournamentMode;
 pub use tournament_status::TournamentStatus;
+pub use tournament_sort_order::TournamentSortOrder;
```
Comment (klautcomputing) [inline shared_types/src/lib.rs:45]: Please set up your IDE to include a newline at the end of the file.

**Review by klautcomputing — CHANGES_REQUESTED** [review]
I have mainly style changes, your Rust looks fine to me.

@IongIer can you take a look at the leptos specifics?

File: apis/src/pages/tournaments.rs [hunk 2]
```diff
@@ -6,56 +6,109 @@ use crate::{
...
+enum TournamentFilter {
+    All,
+    Status(TournamentStatus),
+}
 
 #[component]
 pub fn Tournaments() -> impl IntoView {
     let tournament = expect_context::<TournamentStateContext>();
     let ws = expect_context::<WebsocketContext>();
+    let filter = create_rw_signal(TournamentFilter::Status(TournamentStatus::NotStarted));
```
Comment (IongIer) [inline apis/src/pages/tournaments.rs:23]: these create...functions are getting deprecated. Use RwSignal::new() instead.

File: apis/src/pages/tournaments.rs [hunk 3]
```diff
@@ -6,56 +6,109 @@ use crate::{
...
+                            class=move || get_button_classes(TournamentFilter::Status(TournamentStatus::InProgress), filter.get())
+                            on:click=move |_| filter.set(TournamentFilter::Status(TournamentStatus::InProgress))
+                        >
+                            "In Progress"
+                        </button>
+                        <button
+                            class=move || get_button_classes(TournamentFilter::Status(TournamentStatus::Finished), filter.get())
+                            on:click=move |_| filter.set(TournamentFilter::Status(TournamentStatus::Finished))
+                        >
+                            "Completed"
```
Comment (IongIer) [inline apis/src/pages/tournaments.rs:69]: I would rename this to Finished to keep it consistent with the player profile

**Comment by kd2718** [comment]
Just a note: running `cargo clippy --fix --all-features && leptosfmt apis -q && cargo fmt` changes a ton of files. Maybe we should have a PR to just sync everything

**Review by IongIer — APPROVED** [review]

**Review by klautcomputing — APPROVED** [review]

**Pull request merged**